pkg/volume/fc/attacher.go

```diff
@@ -172,12 +172,27 @@ func (detacher *fcDetacher) UnmountDevice(deviceMountPath string) error {
 	if devName == "" {
 		return nil
 	}
+
 	unMounter := volumeSpecToUnmounter(detacher.mounter, detacher.host)
-	err = detacher.manager.DetachDisk(*unMounter, devName)
+	// The device is unmounted now. If UnmountDevice was retried, GetDeviceNameFromMount
+	// won't find any mount and won't return DetachDisk below.
+	// Therefore implement our own retry mechanism here.
+	// E.g. DetachDisk sometimes fails to flush a multipath device with "device is busy" when it was
+	// just unmounted.
+	var detachError error
+	err = wait.PollImmediate(10*time.Second, time.Minute, func() (bool, error) {
+		detachError = detacher.manager.DetachDisk(*unMounter, devName)
+		if detachError != nil {
+			klog.V(2).Infof("fc: failed to detach disk %s (%s): %v", devName, deviceMountPath, detachError)
```

Comment:
I will probably increase verbosity of the log level here, because for *most* cases - the returned error should capture detach error too.

Reply:
Increased to 4